### PR TITLE
added download report to bpa integration

### PR DESCRIPTION
**Pull request by liornuss-paloalto**
## Status
ready

## Related Issues
fixes https://github.com/demisto/etc/issues/23263

## Description
changed files in BPA integration (Packs/Legacy/Integrations/BPA) to add the feature of downloading a report of BPA from panorama. 
changed the playbook that run the tasks accordingly (Packs/Legacy/Playbooks/playbook-BPA.yml).

- [x] Added the generate_zip_bundle argument to the pan-os-bpa-submit-job command.
       That arg will determine if or not to generate a report of BPA which checks the Panorama. 

- [x] Added the functionality in pan-os-bpa-get-job-results command, that downloads the 
       report in case that it was generated (determined with the arg in pan-os-bpa-submit-job 
       command)

- [x] Added the generate_zip_bundle argument to the Run Panorama Best Practice Assessment 
       Playbook as in input to pan-os-bpa-submit-job task, in order to control whether to 
       generate the report.

## Screenshots

After running  pan-os-bpa-submit-job command with no argument,
when getting the results with pan-os-bpa-get-job-results  command, 
a downloaded report will not be generated.

Screen Shot 2020-05-06 at 19 05 47

After running  pan-os-bpa-submit-job command with arg: generate_zip_bundle=true,
when getting the results with pan-os-bpa-get-job-results  command, 
a downloaded report will be generated and we can download the zip file which contains an html file, pdf file, and excel file.

Screen Shot 2020-05-06 at 19 01 44

## Minimum version of Demisto
- [x] 5.0.0

## Does it break backward compatibility?
- [x] No

## Must have
- [x] Tests
- [ ] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Packs/Legacy/Playbooks/playbook-BPA.yml [hunk 1]
```diff
@@ -47,6 +47,9 @@ tasks:
     nexttasks:
       '#none#':
       - "3"
+    scriptarguments:
+      generate_zip_bundle:
+        simple: "true"
```
Comment (Itay4) [inline Packs/Legacy/Playbooks/playbook-BPA.yml:52]:
```suggestion
        simple: "false"
```

File: Packs/Legacy/Integrations/BPA/BPA.py [hunk 2]
```diff
@@ -181,6 +188,12 @@ def get_results_command(client: Client, args: Dict):
                 checks = get_checks_from_feature(feature_contents[0], feature_name, category_name)
                 job_checks.extend(checks)
 
+    download_url = results.get('download_url')
+
+    # check that a report was generated, and can be downloaded
+    if download_url is not None:
```
Comment (Itay4) [inline Packs/Legacy/Integrations/BPA/BPA.py:194]:
```suggestion
    if download_url:
```

File: Packs/Legacy/Integrations/BPA/BPA.py [hunk 3]
```diff
@@ -101,6 +104,10 @@ def get_results_request(self, task_id: str):
         response = self._http_request('GET', f'results/{task_id}/')
         return response
 
+    def get_download_results_request(self, task_id: str):
```
Comment (Itay4) [inline Packs/Legacy/Integrations/BPA/BPA.py:107]: add return type hint

File: Packs/Legacy/Integrations/BPA/BPA.py [hunk 4]
```diff
@@ -12,6 +12,8 @@
 BPA_HOST = 'https://bpa.paloaltonetworks.com'
 BPA_VERSION = 'v1'
 BPA_URL = BPA_HOST + '/api/' + BPA_VERSION + '/'
+DOWNLOADED_REPORT_NAME = 'panorama-report'
```
Comment (Itay4) [inline Packs/Legacy/Integrations/BPA/BPA.py:15]: why do we need 2 consts for that?

File: Packs/Legacy/Integrations/BPA/BPA.py [hunk 5]
```diff
@@ -191,6 +204,12 @@ def get_results_command(client: Client, args: Dict):
     return human_readable, context, results
 
 
+def download_report_handler(client: Client, task_id):
+    downloaded_report = client.get_download_results_request(task_id)
+    demisto.results(
+        fileResult(DOWNLOADED_REPORT_NAME + DOWNLOADED_REPORT_SUFFIX, downloaded_report, entryTypes['entryInfoFile']))
```
Comment (Itay4) [inline Packs/Legacy/Integrations/BPA/BPA.py:210]: lets add the `task_id` to the file name

File: Packs/Legacy/Playbooks/playbook-BPA.yml [hunk 6]
```diff
@@ -47,6 +47,9 @@ tasks:
     nexttasks:
       '#none#':
       - "3"
+    scriptarguments:
```
Comment (Itay4) [inline Packs/Legacy/Playbooks/playbook-BPA.yml:50]: need to add it to playbook inputs as well

File: Packs/Legacy/Playbooks/playbook-BPA.yml (hunk 1 above)
Comment (Arsenikr) [reply Packs/Legacy/Playbooks/playbook-BPA.yml:52]: yes, set the parameter to false as default

**Pull request merged**